packages/api-server/src/base/gw-config.ts

```diff
@@ -0,0 +1,351 @@
+import { core, utils, HexNumber, Script, HexString } from "@ckb-lumos/base";
+import { normalizers } from "@ckb-lumos/toolkit";
+import {
+  BackendInfo,
+  EoaScript,
+  GwScript,
+  NodeInfo,
+  RollupCell,
+  RollupConfig,
+} from "./types/node-info";
+import {
+  NodeMode,
+  BackendType,
+  EoaScriptType,
+  GwScriptType,
+  GodwokenClient,
+  NodeInfo as GwNodeInfo,
+} from "@godwoken-web3/godwoken";
+import { CKB_SUDT_ID } from "../methods/constant";
+import { Uint32 } from "./types/uint";
+import { snakeToCamel } from "../util";
+
+export class GwConfig {
+  rpc: GodwokenClient;
+  private nodeInfo: NodeInfo | undefined;
+
+  web3ChainId: HexNumber | undefined;
+  accounts: ConfigAccounts | undefined;
+  eoaScripts: ConfigEoaScripts | undefined;
+  backends: ConfigBackends | undefined;
+  gwScripts: ConfigGwScripts | undefined;
+  rollupConfig: RollupConfig | undefined;
+  rollupCell: RollupCell | undefined;
+  nodeMode: NodeMode | undefined;
+  nodeVersion: string | undefined;
+
+  constructor(rpcOrUrl: GodwokenClient | string) {
```

Comment:
```suggestion
  constructor(rpcOrUrl: string) {
```

Seems only `string` used here

Reply:
the last line `this.rpc = rpcOrUrl;` will use GodwokenClient instance, let's keep it?